lib/mongoid/attributes.rb

```diff
@@ -177,8 +177,14 @@ def write_attribute(name, value)
             attribute_will_change!(field_name)
           end
           if localized
-            attributes[field_name] ||= {}
-            attributes[field_name].merge!(typed_value)
+            present = fields[field_name].try(:localize_present?)
+            loc_key, loc_val = typed_value.to_a.flatten
+            if present && loc_val.blank?
+              attributes[field_name].delete(loc_key)
```

Comment:
You need the `||= {}` defaulting here otherwise empty string write as the first value fails:

```
serene2% mongoid=master rails c
Loading development environment (Rails 6.0.5.1)
irb(main):001:0>    class Product
     include Mongoid::Document
     field :description, localize: :present
   end

   I18n.default_locale = :en
   product = Product.new
   product.description = ''

/home/w/apps/mongoid/lib/mongoid/attributes.rb:183:in `block in write_attribute': undefined method `delete' for nil:NilClass (NoMethodError)

              attributes[field_name].delete(loc_key)
```

And this should be added as a test.

Reply:
Or only call `delete` if `attributes[field_name]` is not nil.

Reply:
done